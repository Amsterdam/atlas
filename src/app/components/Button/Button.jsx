import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = ({ title, onClick, children, alignLeft, padding }) => (
  <button
    type="button"
    title={title}
    className={`button-new button-new${(alignLeft) ? '__left' : '__right'}`}
    onClick={onClick}
    style={{
      padding: `${padding}px`
    }}
  >
    {children}
  </button>
);

Button.defaultProps = {
  alignLeft: false,
  children: null,
  padding: 4
};

Button.propTypes = {
  alignLeft: PropTypes.bool,
  children: PropTypes.node,
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  padding: PropTypes.number
};

export default Button;
