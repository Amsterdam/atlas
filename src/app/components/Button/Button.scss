@import "../../../shared/styles/config";

$icon-height: 24px;

.button-new {
  @extend %hover-focus-button;
  background-color: $primary-light;
  border: solid 1px $secondary-gray20;
  box-shadow: 2px 2px 0 0 $secondary-gray40;
  max-height: 34px;
  max-width: 34px;
  padding: 4px;
  position: absolute;
  right: $container-whitespace;
  top: $container-whitespace;
  z-index: 150;

  &__left {
    left: $container-whitespace;
    right: auto;
  }

  .is-embed &,
  .is-embed-preview &,
  .is-print-mode & {
    display: none;
  }
}
