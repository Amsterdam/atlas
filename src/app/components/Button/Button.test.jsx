import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  let component;
  const clickMock = jest.fn();
  beforeEach(() => {
    component = shallow(
      <Button
        title="This is a button"
        onClick={clickMock}
        alignLeft
      >
        Hello, I am a button
      </Button>
    );
  });

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onClick prop when button is clicked', () => {
    component.find('button').simulate('click');
    expect(clickMock).toHaveBeenCalled();
  });
});
