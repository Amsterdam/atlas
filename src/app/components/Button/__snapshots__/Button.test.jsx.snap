// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  className="button-new button-new__left"
  onClick={[MockFunction]}
  style={
    Object {
      "padding": "4px",
    }
  }
  title="This is a button"
  type="button"
>
  Hello, I am a button
</button>
`;
